### PR TITLE
Started manual block election script

### PR DESCRIPTION
**Pull request by pprejs20**

### CODE REVIEW

**Comment by dsiganos** [comment]
But before that, make it work with command arguments and give it support for beta and test networks.

File: manual_block_election.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        with get_connected_socket_endpoint(str(p.ip), p.port) as s:
+            signing_key, verifying_key = node_handshake_id.keypair()
+            node_handshake_id.perform_handshake_exchange(ctx, s, signing_key, verifying_key)
+            resp = get_confirm_block_resp(ctx, genesis_block, s)
+            print("completed")
+            if resp is not None:
+                votes.append(resp)
+            else:
+                continue
+    except OSError:
```
Comment (dsiganos) [inline manual_block_election.py:25]: You want check for both OSError and PyNanoCoinException.

File: manual_block_election.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+votes = []
+
+for p in peers:
+    try:
+        with get_connected_socket_endpoint(str(p.ip), p.port) as s:
+            signing_key, verifying_key = node_handshake_id.keypair()
+            node_handshake_id.perform_handshake_exchange(ctx, s, signing_key, verifying_key)
+            resp = get_confirm_block_resp(ctx, genesis_block, s)
+            print("completed")
```
Comment (miguel1117) [inline manual_block_election.py:20]: We should avoid using print calls in future code. You should use the get_logger function from the _logger module.

File: manual_block_election.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    args = parse_args()
+    ctx = livectx
+
+    if args.beta:
+        ctx = betactx
+    elif args.test:
+        ctx = testctx
+
+    peers = filter(lambda p: p.is_voting, get_peers_from_service(ctx))
+    genesis_block = block_open(ctx['genesis_block']['source'], ctx['genesis_block']['representative'],
```
Comment (dsiganos) [inline manual_block_election.py:36]: This is extremely ugly code, we need something better.

**Comment by dsiganos** [comment]
Let's do confirm by hash (root), and pass in the root of the block to vote on.

**Comment by dsiganos** [comment]
Also, print any relevant votes received for the block, which will be identified by its root.

File: manual_block_election.py [hunk 4]
```diff
@@ -22,7 +22,9 @@ def parse_args():
                        help='use beta network')
     group.add_argument('-t', '--test', action='store_true', default=False,
                        help='use test network')
-    parser.add_argument('-H', '--hash', type=str, required=True)
+    parser.add_argument('-H', '--hash', type=str,
+                        default='991CF190094C00F0B68E2E5F75F6BEE95A2E0BD93CEAA4A6734DB9F19B728948:E89208DD038FBB269987689621D52292AE9C35941A7484756ECCED92A65093BA',
```
Comment (dsiganos) [inline manual_block_election.py:26]: We should not need to make a copy of this, this should be a constant somewhere that we refer to.

Comment (dsiganos) [reply manual_block_election.py:26]: Also, this won;t work for beta and test networks.

File: manual_block_election.py [hunk 5]
```diff
@@ -33,9 +38,8 @@ def main():
         ctx = testctx
 
     peers = filter(lambda p: p.is_voting, get_peers_from_service(ctx))
-    genesis_block = block_open(ctx['genesis_block']['source'], ctx['genesis_block']['representative'],
-                               ctx['genesis_block']['account'], ctx['genesis_block']['signature'],
-                               ctx['genesis_block']['work'])
+    block_hash, root = args.hash.split(':')
```
Comment (dsiganos) [inline manual_block_election.py:43]: Allow the second part of the root to be optional, have a look at confirm_req.py does it

File: constants.py [hunk 6]
```diff
@@ -1 +1,8 @@
+import binascii
+
+import common
+
 max_nano_supply = 133248298 * (10**30)
+genesis_hash_pair_str = common.hash_pair(
```
Comment (dsiganos) [inline constants.py:6]: This cannot possibly work for all network (test/beta)

**Comment by dsiganos** [comment]
What is the status of this PR? I cannot remember.

**Comment by pprejs20** [comment]
> What is the status of this PR? I cannot remember.

Adding genesis blocks to ctx

File: manual_block_election.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import argparse
+import binascii
+
+import _logger
+import common
+from block import block_open
+from confirm_req import get_confirm_hash_resp
+from peercrawler import get_peers_from_service
+from pynanocoin import livectx, get_connected_socket_endpoint, betactx, testctx, get_genesis_block
```
Comment (miguel1117) [inline manual_block_election.py:9]: get_genesis_block does not exist in the pynanocoin module

File: manual_block_election.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            pair = common.hash_pair(binascii.unhexlify(block_hash[0]), binascii.unhexlify(block_hash[1]))
+    else:
+        genesis_block = get_genesis_block(ctx)
+        pair = common.hash_pair(genesis_block.hash(), genesis_block.root())
+
+    votes = []
+    peers_voted = []
+    voting_weights = []
+    rep_map = representative_mapping()
+    rep_map.load_from_file("representative-mappings.json")
```
Comment (miguel1117) [inline manual_block_election.py:57]: Now that we have a representatives service, we should use it instead of this, I think, as this data is not live.

**Review by dsiganos — APPROVED** [review]

**Pull request merged**